distbench_threadpool.cc

```diff
@@ -42,40 +67,114 @@ DistbenchThreadpool::DistbenchThreadpool(int nb_threads) {
   }
 }
 
-DistbenchThreadpool::~DistbenchThreadpool() {
+SimpleThreadpool::~SimpleThreadpool() {
   shutdown_.Notify();
   for (auto& thread : threads_) {
     thread.join();
   }
 }
 
-void DistbenchThreadpool::AddWork(std::function<void()> function) {
+void SimpleThreadpool::AddWork(std::function<void()> function) {
   absl::MutexLock m(&mutex_);
   work_queue_.push(function);
 }
 
-#else
+// This is the C-Threadpool from an external library.
```

Comment:
For my understanding, does this comment mean you copied it from AbstractThreadPool from the external library? If yes, why couldn't you link to it and why did you have to copy it?

Reply:
This comment is just moved around. I want to delete this threadpool from the code. it is ironically not thread safe.